src/librustc_resolve/diagnostics.rs

```diff
@@ -964,6 +964,31 @@ impl<'a> Resolver<'a> {
 
         err.emit();
     }
+
+    crate fn make_undeclared_type_suggestion(
+        &mut self,
+        ident: Ident,
+    ) -> (String, Option<Suggestion>) {
+        let parent_scope = &ParentScope::module(self.graph_root);
+        let typo_suggestion = self.early_lookup_typo_candidate(
+            ScopeSet::AbsolutePath(TypeNS),
+            parent_scope,
+            ident,
+            &|_| true,
```

Comment:
This should use `is_expected` from `PathSource::Type`.
In general, this whole logic should mirror `lookup_typo_candidate` from `fn smart_resolve_report_errors`.